cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -1530,6 +1529,23 @@ class S3FileSystem::Impl : public std::enable_shared_from_this<S3FileSystem::Imp
     }
   }
 
+  // Tests to see if a bucket exists
+  Result<bool> BucketExists(const std::string& bucket) {
+    S3Model::HeadBucketRequest req;
+    req.SetBucket(ToAwsString(bucket));
+
+    auto outcome = client_->HeadBucket(req);
+    if (!outcome.IsSuccess()) {
+      if (!IsNotFound(outcome.GetError())) {
+        return ErrorToStatus(
+            std::forward_as_tuple("When testing for bucket existence '", bucket, "': "),
```

Comment:
"for bucket existence 'xxx'" -> "for existence of bucket 'xxx'" perhaps ?

Reply:
Fixed up